prov/gni/src/gnix_nic.c

```diff
@@ -946,7 +949,13 @@ int gnix_nic_alloc(struct gnix_fid_domain *domain,
 
 	pthread_mutex_lock(&gnix_nic_list_lock);
 
-	if (gnix_nics_per_ptag[domain->ptag] >= gnix_max_nics_per_ptag) {
+	/*
+	 * we can reuse previously allocated nics as long as a
+	 * cdm_id is not specified in the nic_attr arg.
+	 */
+
+	if ((must_alloc_nic == false) && (gnix_nics_per_ptag[domain->ptag] >=
+					 gnix_max_nics_per_ptag)) {
```

Comment:
Could this allow us the provider to create more than max_nics_per_ptag NICs when the called with must_alloc_nic = true?